packages/ember-glimmer/tests/integration/components/fragment-components-test.js

```diff
@@ -165,6 +165,27 @@ moduleFor('Components test: fragment components', class extends RenderingTest {
     this.assertText('baz');
   }
 
+  ['@test does not throw an error if `tagName` is an empty string and `id` is bound property specified via template']() {
+    let template = `{{id}}`;
+    let FooBarComponent = Component.extend({
+      tagName: ''
+    });
+
+    this.registerComponent('foo-bar', { ComponentClass: FooBarComponent, template });
+
+    this.render(`{{#foo-bar id=fooBarId}}{{/foo-bar}}`, { fooBarId: 'baz' });
+
+    this.assertText('baz');
+
+    this.runTask(() => set(this.context, 'fooBarId', 'qux'));
```

Comment:
Can you do a noop rerender before this mutation step?


This should do the trick:

```
this.assertStableRerender();
```